dm/ctl/master/query_status.go

```diff
@@ -60,5 +74,75 @@ func queryStatusFunc(cmd *cobra.Command, _ []string) {
 		return
 	}
 
-	common.PrettyPrintResponse(resp)
+	if resp.Result && taskName == "" && len(workers) == 0 {
+		result := wrapTaskResult(resp)
+		common.PrettyPrintInterface(result)
+	} else {
+		common.PrettyPrintResponse(resp)
+	}
+}
+
+// errorOccurred check ProcessResult and return true if some error occurred
+func errorOccurred(result *pb.ProcessResult) bool {
+	return result != nil && len(result.Errors) > 0
+}
+
+// getRelayStage returns current relay stage (including stageError)
+func getRelayStage(relayStatus *pb.RelayStatus) string {
+	if errorOccurred(relayStatus.Result) {
+		return stageError
+	}
+	return relayStatus.Stage.String()
+}
+
+// wrapTaskResult pick task info and generate tasks' status and relative workers
+func wrapTaskResult(resp *pb.QueryStatusListResponse) *taskResult {
+	taskStatusMap := make(map[string]string, len(resp.Workers))
+	taskCorrespondingWorkers := make(map[string][]string, len(resp.Workers))
+	for _, worker := range resp.Workers {
+		relayStatus := worker.RelayStatus
+		for _, subTask := range worker.SubTaskStatus {
+			subTaskName := subTask.Name
+			subTaskStage := subTask.Stage
+
+			taskCorrespondingWorkers[subTaskName] = append(taskCorrespondingWorkers[subTaskName], worker.Worker)
+			taskStage := taskStatusMap[subTaskName]
+			// the status of a task is decided by its subtasks, the rule is listed as follows:
+			// |                     Subtasks' status                       |                Task's status                 |
+			// | :--------------------------------------------------------: | :------------------------------------------: |
+			// |           Any Paused and len(result.errors) > 0            |    Error - Some error occurred in subtask    |
+			// | Any Running and unit is "Sync" and relay is Paused/Stopped | Error - Relay status is Error/Paused/Stopped |
+			// |              Any Paused but without error                  |                    Paused                    |
+			// |                        All New                             |                     New                      |
+			// |                      All Finished                          |                   Finished                   |
+			// |                      All Stopped                           |                   Stopped                    |
+			// |                         Others                             |                   Running                    |
+			switch {
+			case len(taskStage) >= 5 && taskStage[0:5] == stageError:
```

Comment:
what is the `5` means? need add some comment

Reply:
maybe you can use `strings.HasPrefix`?

Reply:
Revised in https://github.com/pingcap/dm/pull/334/commits/e635e5163c72337b06ce1ec20b66f0b47be3487e